runtime/runtime.go

```diff
@@ -562,15 +562,18 @@ func (rt *Runtime) Serve(ctx context.Context) error {
 		rt.server = rt.server.WithUnixSocketPermission(rt.Params.UnixSocketPerm)
 	}
 
+	ctx, cancel := context.WithCancel(ctx)
 	rt.server, err = rt.server.Init(ctx)
 	if err != nil {
 		rt.logger.WithFields(map[string]interface{}{"err": err}).Error("Unable to initialize server.")
+		cancel()
```

Comment:
My initial reasoning was that `cancel()` should be called before `gracefulServerShutdown()` so that there is time for any active cleanup to complete before the server actually shuts down. But on second thought I don't think it's necessary to have this order of execution. I think `defer cancel()` should do the job.